src/main/scala/sigmastate/eval/CostingRules.scala

```diff
@@ -548,18 +548,20 @@ trait CostingRules extends SigmaLibrary { IR: RuntimeCosting =>
     }
 
     def flatMap[B](fC: RCosted[T => Coll[B]]): RCostedColl[B] = {
+      val sizeF = asCostedFunc[T, Coll[B]](fC).sliceSize
```

Comment:
@greenhat ok, but it is better done after the pattern below is checked.

Reply:
Thanks! Fixed.